core/src/test/java/feign/ResponseTest.java

```diff
@@ -31,30 +31,36 @@
 
   @Test
   public void reasonPhraseIsOptional() {
-    Response response = Response.create(200, null /* reason phrase */, Collections.
-        <String, Collection<String>>emptyMap(), new byte[0]);
+    Response response = Response.builder()
+            .status(200)
+            .reason(null) /* reason phrase */
+            .headers(Collections.<String, Collection<String>>emptyMap())
+            .body(new byte[0])
+            .build();
 
     assertThat(response.reason()).isNull();
     assertThat(response.toString()).isEqualTo("HTTP/1.1 200\n\n");
   }
 
   @Test
   public void lowerCasesNamesOfHeaders() {
-    Response response = Response.create(200,
-                                        null,
-                                        Collections.singletonMap("Content-Type",
-                                                                 Collections.singletonList("application/json")),
-                                        new byte[0]);
+    Response response = Response.builder()
+            .status(200)
+            .reason(null)
+            .headers(Collections.singletonMap("Content-Type", Collections.singletonList("application/json")))
+            .body(new byte[0]).build();
     assertThat(response.headers()).containsOnly(entry(("content-type"), Collections.singletonList("application/json")));
   }
 
   @Test
   public void canAccessHeadersCaseInsensitively() {
     List<String> valueList = Collections.singletonList("application/json");
-    Response response = Response.create(200,
-                                        null,
-                                        Collections.singletonMap("Content-Type", valueList),
-                                        new byte[0]);
+    Response response = Response.builder()
+            .status(200)
+            .reason(null)
+            .headers(Collections.singletonMap("Content-Type", valueList))
```

Comment:
maybe the better way out is to initialize headers to a linkedHashMap and add "putHeader(String, Collection)"? gets you out of generics warfare and cleans up callsites.